File: test-api.js
const http = require('http');

// Test database creation
const testDatabaseCreation = () => {
    const data = JSON.stringify({ name: 'TestDB' });

    const options = {
        hostname: 'localhost',
        port: 8081,
        path: '/api/databases',
        method: 'POST',
        headers: {
            'Content-Type': 'application/json',
            'Content-Length': data.length
        }
    };

    const req = http.request(options, (res) => {
        let body = '';
        res.on('data', (chunk) => {
            body += chunk;
        });
        res.on('end', () => {
            console.log('Database Creation Response:', res.statusCode);
            console.log('Response:', body);
        });
    });

    req.on('error', (e) => {
        console.error('Error:', e);
    });

    req.write(data);
    req.end();
};

// Test using database
const testUseDatabase = () => {
    const data = JSON.stringify({ name: 'TestDB' });

    const options = {
        hostname: 'localhost',
        port: 8081,
        path: '/api/databases/use',
        method: 'POST',
        headers: {
            'Content-Type': 'application/json',
            'Content-Length': data.length
        }
    };

    const req = http.request(options, (res) => {
        let body = '';
        res.on('data', (chunk) => {
            body += chunk;
        });
        res.on('end', () => {
            console.log('Use Database Response:', res.statusCode);
            console.log('Response:', body);
        });
    });

    req.on('error', (e) => {
        console.error('Error:', e);
    });

    req.write(data);
    req.end();
};

// Run tests
console.log('Testing API endpoints...');
testDatabaseCreation();

setTimeout(() => {
    testUseDatabase();
}, 1000);

File: test_api.py
#!/usr/bin/env python3
import requests
import json

def test_api():
    base_url = "http://localhost:8081/api"

    print("Testing API endpoints...")

    # Test 1: Create database
    try:
        response = requests.post(f"{base_url}/databases", json={"name": "TestUI"})
        print(f"Create Database: {response.status_code}")
        print(f"Response: {response.json()}")
    except Exception as e:
        print(f"Create Database Error: {e}")

    # Test 2: Use database
    try:
        response = requests.post(f"{base_url}/databases/use", json={"name": "TestUI"})
        print(f"Use Database: {response.status_code}")
        print(f"Response: {response.json()}")
    except Exception as e:
        print(f"Use Database Error: {e}")

    # Test 3: Create table
    try:
        table_data = {
            "name": "test_table",
            "columns": {
                "id": "INT",
                "name": "VARCHAR(255)",
                "email": "VARCHAR(255)"
            }
        }
        response = requests.post(f"{base_url}/tables", json=table_data)
        print(f"Create Table: {response.status_code}")
        print(f"Response: {response.json()}")
    except Exception as e:
        print(f"Create Table Error: {e}")

    print("API testing completed!")

if __name__ == "__main__":
    test_api()